packages/flutter_tools/test/general.shard/ios/native_assets_test.dart

```diff
@@ -0,0 +1,242 @@
+    await packageConfig.create();
+    final Uri? nativeAssetsYaml = await dryRunNativeAssetsiOS(
+      projectUri: projectUri,
+      fileSystem: fileSystem,
+      buildRunner: FakeNativeAssetsBuildRunner(
+        packagesWithNativeAssetsResult: <Package>[
+          Package('bar', projectUri),
+        ],
+        dryRunResult: FakeNativeAssetsBuilderResult(
+          assets: <Asset>[
+          Asset(
+            id: 'package:bar/bar.dart',
+            linkMode: LinkMode.dynamic,
+            target: native_assets_cli.Target.macOSArm64,
+            path: AssetAbsolutePath(Uri.file('bar.dylib')),
+          ),
+          Asset(
+            id: 'package:bar/bar.dart',
+            linkMode: LinkMode.dynamic,
+            target: native_assets_cli.Target.macOSX64,
+            path: AssetAbsolutePath(Uri.file('bar.dylib')),
+          ),
+          ],
+        ),
+      ),
+    );
+    expect(
+      nativeAssetsYaml,
+      projectUri.resolve('build/native_assets/ios/native_assets.yaml'),
+    );
+    expect(
+      await fileSystem.file(nativeAssetsYaml).readAsString(),
+      contains('package:bar/bar.dart'),
+    );
+  });
+
+  testUsingContext('build with assets but not enabled', () async {
+    final File packageConfig =
+        environment.projectDir.childFile('.dart_tool/package_config.json');
+    await packageConfig.parent.create();
+    await packageConfig.create();
+    expect(
+      () => buildNativeAssetsiOS(
+        darwinArchs: <DarwinArch>[DarwinArch.arm64],
+        environmentType: EnvironmentType.simulator,
+        projectUri: projectUri,
+        buildMode: BuildMode.debug,
+        fileSystem: fileSystem,
+        writeYamlFileTo: environment.buildDir.uri,
+        buildRunner: FakeNativeAssetsBuildRunner(
+          packagesWithNativeAssetsResult: <Package>[
+            Package('bar', projectUri),
+          ],
+        ),
+      ),
+      throwsToolExit(
+        message:
+            'Package(s) bar require the native assets feature to be enabled. '
+            'Enable using `flutter config --enable-native-assets`.',
+      ),
+    );
+  });
+
+  testUsingContext('build no assets', overrides: <Type, Generator>{
+    FeatureFlags: () => TestFeatureFlags(isNativeAssetsEnabled: true),
+    ProcessManager: () => FakeProcessManager.any(),
+  }, () async {
+    final File packageConfig =
+        environment.projectDir.childFile('.dart_tool/package_config.json');
+    await packageConfig.parent.create();
+    await packageConfig.create();
+    await buildNativeAssetsiOS(
+      darwinArchs: <DarwinArch>[DarwinArch.arm64],
+      environmentType: EnvironmentType.simulator,
+      projectUri: projectUri,
+      buildMode: BuildMode.debug,
+      fileSystem: fileSystem,
+      writeYamlFileTo: environment.buildDir.uri,
+      buildRunner: FakeNativeAssetsBuildRunner(
+        packagesWithNativeAssetsResult: <Package>[
+          Package('bar', projectUri),
+        ],
+      ),
+    );
```

Comment:
You don't have any expectations for this test? Wouldn't you expect the returned list of Uris to be empty, a yaml file created in the file system, no assets created, etc?